application/models/services/SurveyCondition.php

```diff
@@ -0,0 +1,1639 @@
+<?php
+
+namespace LimeSurvey\Models\Services;
+
+use LimeSurvey\Models\Services\Exception\PermissionDeniedException;
+use LSYii_Application;
+use Permission;
+use Survey;
+use Response;
+
+class SurveyCondition
+{
+    private LSYii_Application $app;
+    private Permission $permission;
+    private Survey $survey;
+    protected int $iSurveyID;
+    protected bool $tokenTableExists;
+    protected array $tokenFieldsAndNames;
+    protected string $language;
+    protected const X = 'X';
+
+    /**
+     * Sets the iSurveyID member of this object
+     * @param int $iSurveyID
+     * @return static
+     */
+    public function setISurveyID(int $iSurveyID)
+    {
+        $this->iSurveyID = $iSurveyID;
+        return $this;
+    }
+
+    /**
+     * Gets the survey table by name and id
+     * @param string $name
+     * @param int $id
+     * @return string
+     */
+    public function getSurveyTable(string $name, int $id)
+    {
+        switch ($name) {
+            case 'token':
+                return "{{tokens_$id}}";
+            default:
+                return '';
+        }
+    }
+
+    /**
+     * Gets the question's id from the field's name
+     * @param string $copyc
+     * @return string
+     */
+    public function getQIDFromFieldName(string $copyc)
+    {
+        list(,, $newqid) = explode("X", (string) $copyc);
+        return $newqid;
+    }
+
+    /**
+     * Gets the field's name by sid, gid, qid and title
+     * @param mixed $sid
+     * @param mixed $gid
+     * @param mixed $qid
+     * @param mixed $title
+     * @return string
+     */
+    public function getFieldName($sid, $gid, $qid, $title = '')
+    {
+        return $sid . self::X . $gid . self::X . $qid . $title;
+    }
+
+    public function __construct(
```

Comment:
By convention constructor should be the first method on the class.

Reply:
@kevin-foster-uk applied the change